server/queue/queue-pulsar/src/main/scala/org/apache/james/queue/pulsar/PulsarMailQueue.scala

```diff
@@ -110,11 +108,8 @@ class PulsarMailQueue(
   private implicit val implicitSystem: ActorSystem = system
   private implicit val ec: ExecutionContextExecutor = system.dispatcher
   private implicit val implicitBlobIdFactory: BlobId.Factory = blobIdFactory
-  private implicit val client: PulsarAsyncClient = PulsarClient(config.pulsar.brokerUri)
-  private val admin = {
-    val builder = PulsarAdmin.builder()
-    builder.serviceHttpUrl(config.pulsar.adminUri).build()
-  }
+  private implicit val client: PulsarAsyncClient = config.pulsar.asyncClient()
+  private val admin = config.pulsar.adminClient()
 
```

Comment:
Shouldn't the pulsar client be a global entity rather than a "per queue" entity?

 -> Sounds like expensive resources to maintain
 -> the chances of the leaks increases

My opinion is that it would better fit as a backend-pulsar component injected down the stack.

Reply:
We will have to check all the implications but this sounds reasonable to me. Would you accept a `lazy val` on `pulsarConfiguration` to replace the `def` for `asyncClient` and `adminClient` respectivey ?

Reply:
Shouldnt a configuration be a dumb pojo?

Reply:
we reworked the code to match your suggestion